core/src/main/scala/cats/std/string.scala

```diff
@@ -1,7 +1,24 @@
 package cats
 package std
 
-trait StringInstances extends algebra.std.StringInstances {
+trait StringInstances {
+  implicit val stringOrder: Order[String] = new StringOrder
+  implicit val stringMonoid = new StringMonoid
   implicit val stringShow: Show[String] =
     Show.fromToString[String]
 }
+
+class StringOrder extends Order[String] {
```

Comment:
I'm wondering if we should make many of the classes in this PR, such as `StringOrder` `private[cats]`. This would follow the precedent set in #612, and I think it's especially important considering the binary compatibility constraints on the `kernel` module. If people find themselves really wanting access to these classes, then we can always open them for extension, but it's much easier to do that then it is to make changes once they have been exposed as part of the API.